### PR TITLE
Security Self-review

### PR DESCRIPTION
**Pull request by fivedots**
A security self-review is required to start a TAG review.

### CODE REVIEW

File: security-self-review.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+This specification does not allow access to sensors.
+
+## 2.8. What data does this specification expose to an origin? Please also document what data is identical to data exposed by other features, in the same or different contexts.
+
+The data exposed is only the one that was previously created by the origin. This
+is similar to other storage APIs
+
+## 2.9. Does this specification enable new script execution/loading mechanisms?
+
+This specification does not enable new script execution/loading mechanisms.
```
Comment (rstz) [inline security-self-review.md:46]: Nit: Below, you omitted the newline before the next heading

**Review by rstz — APPROVED** [review]

File: security-self-review.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+## 2.4. How does this specification deal with sensitive information?
+
+This specification does not deal with sensitive information.
+
+## 2.5. Does this specification introduce new state for an origin that persists across browsing sessions?
+
+Yes, this specification introduces a new persistent storage mechanism where
+state may be kept. The risk of tracking must be mitigated by following similar
+strategies used in other storage APIs e.g. user agents should allow users to
+wipe stored data, an origin may only access files that are owned by it, etc.
```
Comment (mikewest) [inline security-self-review.md:27]: "..., access in third-party contexts should be governed by the same policy as other storage types, etc."

File: security-self-review.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## 2.6. What information from the underlying platform, e.g. configuration data, is exposed by this specification to an origin?
+
+No information from the underlying platform is exposed. However care must be
+taken during implementation, such as not to expose operating-system-specific
+behavior through the way file operations are executed.
```
Comment (mikewest) [inline security-self-review.md:33]: Maybe point this out in a "Security Considerations" section of the eventual spec?

File: security-self-review.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+This specification does not allow any control of a user agent’s UI.
+## 2.12. What temporary identifiers might this specification create or expose to the web?
+
+This specification does not expose any temporary identifiers.
+## 2.13. How does this specification distinguish between behavior in first-party and third-party contexts?
+
+This specification does not offer different behaviors depending on the context.
+It’s worth noting that the data accessible to each context will depend on the
+origin, and therefore a call from a third-party context would not be able to
+access first-party content.
```
Comment (mikewest) [inline security-self-review.md:61]: As above, I'd suggest reframing this in terms of NativeIO following the same policies a user agents wishes to place on any other kind of storage. It should be indistinguishable from `localStorage` from that perspective.

File: security-self-review.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+## 2.14. How does this specification work in the context of a user agent’s Private Browsing or "incognito" mode?
+
+This specification will work mostly the same as in “regular” mode, except that
+stored information will not be persisted after a private session is finished.
```
Comment (mikewest) [inline security-self-review.md:66]: "This should be indistinguishable from other storage APIs."

**Review by mikewest — APPROVED** [review]
LGTM % small comments.

File: security-self-review.md (hunk 1 above)
Comment (fivedots) [reply security-self-review.md:46]: Done!

File: security-self-review.md (hunk 2 above)
Comment (fivedots) [reply security-self-review.md]: Done

File: security-self-review.md (hunk 3 above)
Comment (fivedots) [reply security-self-review.md:35]: Thanks, added it to the explainer for now!

File: security-self-review.md (hunk 4 above)
Comment (fivedots) [reply security-self-review.md]: Done.

File: security-self-review.md (hunk 5 above)
Comment (fivedots) [reply security-self-review.md:72]: Done.

**Pull request merged**